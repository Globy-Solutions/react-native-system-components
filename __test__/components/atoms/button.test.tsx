import {fireEvent, screen} from '@testing-library/react-native';

import {Button} from '../../../';
import {render} from '../../utils/customRender';

const onPressMock = jest.fn();
const defaultProps = {
  onPress: onPressMock,
  title: 'Test Button'
};
export const Btn = () => <Button {...defaultProps} />;

describe('Button', () => {
  let buttonElement: any;
  beforeEach(() => {
    jest.clearAllMocks();
    global.fetch = jest.fn().mockResolvedValue({
      json: () => Promise.resolve([])
    });
    const {getByText} = render(<Btn />);
    buttonElement = getByText(defaultProps.title);
  });
  test('Match Snapshot', () => {
    expect(screen).toMatchSnapshot();
  });

  test('renders correctly', () => {
    expect(buttonElement).toBeTruthy();
  });
  test('calls onPress function when pressed', () => {
    fireEvent.press(buttonElement);
    expect(onPressMock).toHaveBeenCalled();
  });
});
