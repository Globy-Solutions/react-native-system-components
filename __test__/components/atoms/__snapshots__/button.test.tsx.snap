// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Match Snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderColor": "black",
        "borderRadius": 10,
        "borderWidth": 1,
        "elevation": 5,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Test Button
  </Text>
</View>
`;
